### PR TITLE
AKU-819: Non YUI2 create/edit site dialogs

### PR DESCRIPTION
**Pull request by draperd**
This PR addresses https://issues.alfresco.com/jira/browse/AKU-819 to make the SiteService configurable so that it can use Aikau dialogs for creating and editing sites. The unit tests have been updated and this has been manually tested against Share. The default behaviour is still to attempt to use the Share specific YUI2 dialogs.

### CODE REVIEW

File: aikau/src/main/resources/alfresco/core/topics.js [hunk 1]
```diff
@@ -993,6 +1018,62 @@ define([],function() {
       SET_THUMBNAIL_SIZE: "ALF_SET_THUMBNAIL_SIZE",
 
       /**
+       * This topic can be published to perform the actual creation of a site. Unlike 
+       * [CREATE_SITE]{@link module:alfresco/core/topics#CREATE_SITE} this requires a payload that includes the
+       * details for the new site (as it performs the actual creation on the Alfresco Repository rather than
```
Comment (MartinDoyleUK) [inline aikau/src/main/resources/alfresco/core/topics.js]: What are the details? No properties listed.

File: aikau/src/main/resources/alfresco/core/topics.js [hunk 2]
```diff
@@ -993,6 +1018,62 @@ define([],function() {
...
+       * @since 1.0.55
+       *
+       * @event
+       */
+      SITE_CREATION_SUCCESS: "ALF_SITE_CREATION_SUCCESS",
+
+      /**
+       * This topic can be published to perform the actual creation of a site. Unlike 
+       * [CREATE_SITE]{@link module:alfresco/core/topics#CREATE_SITE} this requires a payload that includes the
+       * details for the new site (as it performs the actual creation on the Alfresco Repository rather than
```
Comment (MartinDoyleUK) [inline aikau/src/main/resources/alfresco/core/topics.js]: Properties missing again?

File: aikau/src/main/resources/alfresco/services/SiteService.js [hunk 3]
```diff
@@ -46,6 +47,34 @@ define(["dojo/_base/declare",
...
+       * This is an array of site preset options that will be displayed when creating a site (when the service
+       * is not configured to be in [legacyMode]{@link module:alfresco/services/SiteService#legacyMode}). The
+       * default configuration just contains the "Collaboration Site", but this can be re-configured to contain
+       * additional site presets. Any additional presets must have values that map to presets in the Surf configuration.
+       * 
+       * @instance
+       * @type {object[]}
+       * @since 1.0.55
+       */
+      sitePresets: [
```
Comment (MartinDoyleUK) [inline aikau/src/main/resources/alfresco/services/SiteService.js]: This is static ... presuming this is ok.

File: aikau/src/main/resources/alfresco/services/SiteService.js [hunk 4]
```diff
@@ -593,52 +627,275 @@ define(["dojo/_base/declare",
...
+       * [widgetsForEditSiteDialog]{@link module:alfresco/services/SiteService#widgetsForEditSiteDialog}.
+       *
+       * @instance
+       * @param {object} response The details of the site to be edited
+       * @param {object} originalRequestConfig The details of the request made to obtain site data
+       * @fires module:alfresco/core/topics#CREATE_FORM_DIALOG
+       */
+      showEditSiteDialog: function alfresco_services_SiteService__showEditSiteDialog(response, originalRequestConfig) {
+         // Check that the resposne is the expected siteData...
+         var shortName = lang.getObject("shortName", false, response);
```
Comment (MartinDoyleUK) [inline aikau/src/main/resources/alfresco/services/SiteService.js]: Should this be inside the `response` condition?

File: aikau/src/main/resources/alfresco/services/SiteService.js [hunk 5]
```diff
@@ -593,52 +627,275 @@ define(["dojo/_base/declare",
...
+       * @instance
+       * @param  {object} response The response from the request to edit the site
+       * @param  {object} originalRequestConfig The configuration for the request to edit the site
+       * @since 1.0.55
+       * @fires module:alfresco/core/topics#SITE_CREATION_SUCCESS
+       * @fires module:alfresco/core/topics#NAVIGATE_TO_PAGE
+       */
+      onSiteEditSuccess: function alfresco_services_SiteService__onSiteEditSuccess(/*jshint unused:false*/ response, originalRequestConfig) {
+         this.alfPublish(topics.SITE_EDIT_SUCCESS);
+         this.alfPublish(topics.NAVIGATE_TO_PAGE, {
```
Comment (MartinDoyleUK) [inline aikau/src/main/resources/alfresco/services/SiteService.js]: `this.alfServicePublish`?

File: aikau/src/main/resources/alfresco/services/SiteService.js [hunk 6]
```diff
@@ -818,6 +1075,199 @@ define(["dojo/_base/declare",
...
+               },
+               validationConfig: [
+                  {
+                     validation: "maxLength",
+                     length: 72,
+                     errorMessage: "create-site.dialog.urlname.maxLength"
+                  },
+                  {
+                     validation: "regex",
+                     regex: "^[0-9a-zA-Z]+$",
```
Comment (MartinDoyleUK) [inline aikau/src/main/resources/alfresco/services/SiteService.js]: According to the error message, this should allow hyphens as well ...

File: aikau/src/test/resources/config/Suites.js [hunk 7]
```diff
@@ -31,8 +31,8 @@ define({
     * @type {string[]}
     */
    // Uncomment and add specific tests as necessary during development!
-   xbaseFunctionalSuites: [
-      "src/test/resources/alfresco/upload/UploadMonitorTest"
+   baseFunctionalSuites: [
```
Comment (MartinDoyleUK) [inline aikau/src/test/resources/config/Suites.js]: Forgot to undo this?

File: aikau/src/main/resources/alfresco/services/SiteService.js [hunk 8]
```diff
@@ -593,52 +627,275 @@ define(["dojo/_base/declare",
...
+       * @instance
+       * @param  {object} response The response from the request to create the site
+       * @param  {object} originalRequestConfig The configuration for the request to create the site
+       * @since 1.0.55
+       * @fires module:alfresco/core/topics#SITE_CREATION_SUCCESS
+       * @fires module:alfresco/core/topics#NAVIGATE_TO_PAGE
+       */
+      onSiteCreationSuccess: function alfresco_services_SiteService__onSiteCreationSuccess(/*jshint unused:false*/ response, originalRequestConfig) {
+         this.alfPublish(topics.SITE_CREATION_SUCCESS);
+         this.alfPublish(topics.NAVIGATE_TO_PAGE, {
```
Comment (MartinDoyleUK) [inline aikau/src/main/resources/alfresco/services/SiteService.js]: `this.alfServicePublish`?

File: aikau/src/main/resources/alfresco/services/SiteService.js (hunk 4 above)
Comment (draperd) [inline aikau/src/main/resources/alfresco/services/SiteService.js]: No, I was using this to just check that the response maps to the site data as best as I could.

**Comment by draperd** [comment]
@andyhealey26 This PR features messages

**Pull request merged**